File: app/src/main/java/com/dev/hms/MainActivity.java
package com.dev.hms;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import androidx.appcompat.app.AppCompatActivity;

import com.google.firebase.auth.FirebaseAuth;

public class MainActivity extends AppCompatActivity {
    private FirebaseAuth firebaseAuth;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        firebaseAuth=FirebaseAuth.getInstance();

        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {

                connexion();
            }
        }, 2000);
    }

    private void connexion() {
        if (firebaseAuth.getCurrentUser()!=null){
            Intent intent =new Intent(MainActivity.this,home.class);
            startActivity(intent);


        }else {
            Intent intent =new Intent(MainActivity.this,login.class);
            startActivity(intent);

        }
    }
}

File: app/src/main/java/com/dev/hms/MainActivity.kt
package com.dev.hms

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.view.View
import android.view.Window
import android.view.WindowManager
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.auth.FirebaseAuth


class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        window.setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN)
        setContentView(R.layout.activity_main)

        //4second splash time
        Handler().postDelayed({
            //start main activity
            startActivity(Intent(this, home::class.java))
            //finish this activity
            finish()
        },4000)
    }

}